src/Features/CSharp/Portable/PullMemberUp/CSharpMembersPullerService.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.  
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    [ExportLanguageService(typeof(IMembersPullerService), LanguageNames.CSharp), Shared]
+    internal class CSharpMembersPullerService : AbstractMembersPullerService<UsingDirectiveSyntax>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpMembersPullerService()
+        {
+        }
+
+        protected override ImmutableArray<UsingDirectiveSyntax> GetImports(SyntaxNode node)
+        {
+            return node.AncestorsAndSelf()
+                .Where(node => node is CompilationUnitSyntax || node is NamespaceDeclarationSyntax)
```

Comment:
we're already generic on UsingDirectiveSyntax.  Consider making generic on these two as well, having the base class do the walk up, and the call into the derived just to grab the directices.